tests/integration_tests/conftest.py

```diff
@@ -326,7 +326,8 @@ def virtual_dataset():
     TableColumn(column_name="col5", type="VARCHAR(255)", table=dataset)
 
     SqlMetric(metric_name="count", expression="count(*)", table=dataset)
-    db.session.merge(dataset)
+    db.session.add(dataset)
+    db.session.commit()
```

Comment:
There's likely a lot wrong happening here, but committing seems consistent with the logic on lines #334-335.